src/java/voldemort/store/readonly/ReadOnlyStorageEngine.java

```diff
@@ -535,6 +534,10 @@ public void truncate() {
             } else {
                 return Collections.emptyList();
             }
+        } catch (IllegalStateException e) {
+            logger.warn("ChunkedFileSet.getChunkForKey() did not execute successfully. " +
+                        "Returning empty result for key: " + ByteUtils.toHexString(key.get()), e);
```

Comment:
Yeah totally. I actually thought about doing that and then got distracted. I'll add it in.